diff --git a/dist/ts-eslint/eslint/FlatESLint.js b/dist/ts-eslint/eslint/FlatESLint.js
index 37f84a2e39a943d9d97a746d0814bcba0f13749d..d87a0c8a0567b6a17884e1d0df1ad282cc20e985 100644
--- a/dist/ts-eslint/eslint/FlatESLint.js
+++ b/dist/ts-eslint/eslint/FlatESLint.js
@@ -9,6 +9,6 @@ const use_at_your_own_risk_1 = require("eslint/use-at-your-own-risk");
  *
  * If you want to lint code on browsers, use the Linter class instead.
  */
-class FlatESLint extends use_at_your_own_risk_1.FlatESLint {
+class FlatESLint extends (use_at_your_own_risk_1.FlatESLint ?? require("eslint").ESLint) {
 }
 exports.FlatESLint = FlatESLint;
diff --git a/dist/ts-eslint/eslint/LegacyESLint.js b/dist/ts-eslint/eslint/LegacyESLint.js
index f40012bf3c378d3f8d2c6d0f8d7be8845bab619d..76c01e2cbcc42127cabadce270a073a890c818a0 100644
--- a/dist/ts-eslint/eslint/LegacyESLint.js
+++ b/dist/ts-eslint/eslint/LegacyESLint.js
@@ -9,6 +9,6 @@ const use_at_your_own_risk_1 = require("eslint/use-at-your-own-risk");
  *
  * If you want to lint code on browsers, use the Linter class instead.
  */
-class LegacyESLint extends use_at_your_own_risk_1.LegacyESLint {
+class LegacyESLint extends (use_at_your_own_risk_1.LegacyESLint ?? require("eslint").ESLint) {
 }
 exports.LegacyESLint = LegacyESLint;
