diff --git a/eslint-plugin-prettier.js b/eslint-plugin-prettier.js
index 46cb34484c1d26b7fc1bf06c0840bdbf77d6c774..efaec258f972b80530e7ce268a3ea65b2732cc82 100644
--- a/eslint-plugin-prettier.js
+++ b/eslint-plugin-prettier.js
@@ -18,6 +18,7 @@
  *   parserMeta?: ESLint.ObjectMetaProperties['meta'];
  *   parserPath?: string;
  *   usePrettierrc?: boolean;
+ *   eslintTakeoverMode?: boolean;
  * }} Options
  *
  *
@@ -25,6 +26,7 @@
  *   source: string,
  *   options: Options,
  *   fileInfoOptions: FileInfoOptions,
+ *   eslintTakeoverMode: boolean,
  * ) => string} PrettierFormat
  */
 
@@ -165,6 +167,7 @@ const eslintPluginPrettier = {
                 properties: {},
                 additionalProperties: true,
               },
+              eslintTakeoverMode: { type: 'boolean', description: "Use ESLint parser and ignore list instead of Prettier's. This allows this plugin to be used to [lint fenced code blocks inside Markdown](https://github.com/prettier/eslint-plugin-prettier/pull/613). The original author is Anthony Fu (@antfu)." },
             },
             additionalProperties: true,
           },
@@ -180,6 +183,7 @@ const eslintPluginPrettier = {
         const usePrettierrc = !options || options.usePrettierrc !== false;
         /** @type {FileInfoOptions} */
         const fileInfoOptions = options?.fileInfoOptions || {};
+        const eslintTakeoverMode = options?.eslintTakeoverMode ?? false;
 
         // `context.getSourceCode()` was deprecated in ESLint v8.40.0 and replaced
         // with the `sourceCode` property.
@@ -248,6 +252,7 @@ const eslintPluginPrettier = {
                   usePrettierrc,
                 },
                 fileInfoOptions,
+                eslintTakeoverMode,
               );
             } catch (err) {
               if (!(err instanceof SyntaxError)) {
diff --git a/worker.mjs b/worker.mjs
index b76274ce25cfcd59bd8004ed59e2329e8da6beb8..01bf8fb7d6f06593d0f7aab14d984c20014898a6 100644
--- a/worker.mjs
+++ b/worker.mjs
@@ -32,6 +32,7 @@ runAsWorker(
       ...eslintPrettierOptions
     },
     eslintFileInfoOptions,
+    eslintTakeoverMode = false,
   ) => {
     const prettierRcOptions = usePrettierrc
       ? await prettier.resolveConfig(onDiskFilepath, {
@@ -39,6 +40,9 @@ runAsWorker(
         })
       : null;
 
+    const initialOptions = {};
+
+    if(!eslintTakeoverMode) {
     const { ignored, inferredParser } = await prettier.getFileInfo(
       onDiskFilepath,
       {
@@ -57,7 +61,7 @@ runAsWorker(
       return;
     }
 
-    const initialOptions = { parser: inferredParser ?? 'babel' };
+    initialOptions.parser = inferredParser ?? 'babel';
 
     // ESLint supports processors that let you extract and lint JS
     // fragments within a non-JS language. In the cases where prettier
@@ -172,6 +176,7 @@ runAsWorker(
         return;
       }
     }
+    }
 
     /** @type {PrettierOptions} */
     const prettierOptions = {
