diff --git a/eslint-plugin-prettier.js b/eslint-plugin-prettier.js
index 7a6ecedaf60013f14308f636dfc031e0647fe099..5a0c3c110645da1258fce1e3c4462e4b0ab3d0c7 100644
--- a/eslint-plugin-prettier.js
+++ b/eslint-plugin-prettier.js
@@ -18,6 +18,7 @@
  *   parserMeta?: ESLint.ObjectMetaProperties['meta'];
  *   parserPath?: string;
  *   usePrettierrc?: boolean;
+ *   eslintTakeoverMode?: boolean;
  * }} Options
  *
  *
@@ -25,6 +26,7 @@
  *   source: string,
  *   options: Options,
  *   fileInfoOptions: FileInfoOptions,
+ *   eslintTakeoverMode: boolean,
  * ) => string} PrettierFormat
  */
 
@@ -167,6 +169,7 @@ const eslintPluginPrettier = {
                 properties: {},
                 additionalProperties: true,
               },
+              eslintTakeoverMode: { type: 'boolean', description: "Use ESLint parser and ignore list instead of Prettier's. This allows this plugin to be used to [lint fenced code blocks inside Markdown](https://github.com/prettier/eslint-plugin-prettier/pull/613). The original author is Anthony Fu (@antfu)." },
             },
             additionalProperties: false,
           },
@@ -182,6 +185,7 @@ const eslintPluginPrettier = {
         const usePrettierrc = !options || options.usePrettierrc !== false;
         /** @type {FileInfoOptions} */
         const fileInfoOptions = options?.fileInfoOptions || {};
+        const eslintTakeoverMode = options?.eslintTakeoverMode ?? false;
 
         // `context.getSourceCode()` was deprecated in ESLint v8.40.0 and replaced
         // with the `sourceCode` property.
@@ -250,6 +254,7 @@ const eslintPluginPrettier = {
                   usePrettierrc,
                 },
                 fileInfoOptions,
+                eslintTakeoverMode,
               );
             } catch (err) {
               if (!(err instanceof SyntaxError)) {
diff --git a/worker.mjs b/worker.mjs
index 1451209f44b626ef3e6f0b8c3e719db99fb791f8..ceb3b4bb8049ecdb8714a358126e91fcdf1b0c53 100644
--- a/worker.mjs
+++ b/worker.mjs
@@ -32,6 +32,7 @@ runAsWorker(
       ...eslintPrettierOptions
     },
     eslintFileInfoOptions,
+    eslintTakeoverMode = false,
   ) => {
     const prettierRcOptions = usePrettierrc
       ? await prettier.resolveConfig(onDiskFilepath, {
@@ -39,6 +40,9 @@ runAsWorker(
         })
       : null;
 
+    const initialOptions = {};
+
+    if(!eslintTakeoverMode) {
     const { ignored, inferredParser } = await prettier.getFileInfo(
       onDiskFilepath,
       {
@@ -55,7 +59,7 @@ runAsWorker(
       return;
     }
 
-    const initialOptions = { parser: inferredParser ?? 'babel' };
+    initialOptions.parser = inferredParser ?? 'babel';
 
     // ESLint supports processors that let you extract and lint JS
     // fragments within a non-JS language. In the cases where prettier
@@ -173,6 +177,7 @@ runAsWorker(
         return;
       }
     }
+    }
 
     /** @type {PrettierOptions} */
     const prettierOptions = {
