diff --git a/dist/core.mjs b/dist/core.mjs
index febff6bdfebd194ad10c163c0a092d0aedc6ae0a..d2cca959ae0176866a42ccd095ec4c3148e68cd5 100644
--- a/dist/core.mjs
+++ b/dist/core.mjs
@@ -34,7 +34,7 @@ async function pluginsToRulesDTS(plugins, options = {}) {
   const rules = [];
   for (const [pluginName, plugin] of Object.entries(plugins)) {
     for (const [ruleName, rule] of Object.entries(plugin.rules || {})) {
-      if (rule?.meta) {
+      if (rule?.meta == null || typeof rule.meta === 'object') {
         rules.push([
           pluginName ? `${pluginName}/${ruleName}` : ruleName,
           rule
