diff --git a/dist/index.js b/dist/index.js
index cdf35fb57eadd8b1711866d0b7b60dfe417c5ef1..01cb6a1615a123b1e7c1c6583cff5f38bc95197c 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3061,7 +3061,7 @@ var require_FlatESLint = /* @__PURE__ */ __commonJS({ "node_modules/.pnpm/@types
 	*
 	* If you want to lint code on browsers, use the Linter class instead.
 	*/
-	var FlatESLint = class extends use_at_your_own_risk_1$1.FlatESLint {};
+	var FlatESLint = class extends (use_at_your_own_risk_1$1.FlatESLint ?? __require("eslint").ESLint) {};
 	exports.FlatESLint = FlatESLint;
 }) });
 
@@ -3077,7 +3077,7 @@ var require_LegacyESLint = /* @__PURE__ */ __commonJS({ "node_modules/.pnpm/@typ
 	*
 	* If you want to lint code on browsers, use the Linter class instead.
 	*/
-	var LegacyESLint = class extends use_at_your_own_risk_1.LegacyESLint {};
+	var LegacyESLint = class extends (use_at_your_own_risk_1.LegacyESLint ?? __require("eslint").ESLint) {};
 	exports.LegacyESLint = LegacyESLint;
 }) });
 
