diff --git a/dist/index.js b/dist/index.js
index 04e2b0bfb75a5593bac7dfc1af70ff917069d79e..d753834937b514495f5e20fe6945c28ec0c05262 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -170,7 +170,7 @@ const P = G(), K = P?.name || "";
 */
 function isInEditor() {
 	if (T) return false;
-	return toBoolean(o.VSCODE_PID) || toBoolean(o.VSCODE_CWD) || toBoolean(o.JETBRAINS_IDE) || toBoolean(o.VIM) || toBoolean(o.NVIM) || toBoolean(o.ZED_IMPERSONATE) || toBoolean(o.EMACS) || toBoolean(o.ELLSP_EMACS) || toBoolean(o.INSIDE_EMACS) || toBoolean(o.EASK_EMACS) || o.TERM_PROGRAM === "vscode";
+	return toBoolean(o.VSCODE_PID) || toBoolean(o.VSCODE_CWD) || toBoolean(o.JETBRAINS_IDE) || toBoolean(o.VIM) || toBoolean(o.NVIM) || toBoolean(o.ZED_IMPERSONATE) || toBoolean(o.EMACS) || toBoolean(o.ELLSP_EMACS) || toBoolean(o.INSIDE_EMACS) || toBoolean(o.EASK_EMACS)/*  || o.TERM_PROGRAM === "vscode" */;
 }
 /**
 * @see https://github.com/unjs/std-env/blob/main/src/_utils.ts
