diff --git a/build/index.js b/build/index.js
index b241b7e347f16ee76b08f1ba7f54d3f2ee208267..d3e6206305b023ec2f8b0df59765eb6f31137777 100644
--- a/build/index.js
+++ b/build/index.js
@@ -41565,9 +41565,9 @@
   }), h.render());
   async function a(u) {
     const l = $i(e[u]), [f, o] = l || [u, e[u]], c = t.target || "latest", d = typeof c == "string" ? c : c(f, (0, Et.parseRange)(o)), [p, m] = d.startsWith("@") ? ["distTag", d.slice(1)] : [d, "latest"];
-    if (!t.cooldown) {
+    {
       const w = t.cacher?.get(f, p);
-      if (w?.version && (w?.time || !t.format?.includes("time")))
+      if (w?.version && (w?.time || !t.format?.includes("time")) && (!t.cooldown || w?.time))
         return h?.tick(), w;
     }
     let g;
